vector/src/main/java/im/vector/adapters/NotificationTroubleshootRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,171 @@
+package im.vector.adapters
+
+import android.content.res.ColorStateList
+import android.os.Build
+import android.support.transition.TransitionManager
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.TextView
+import im.vector.R
+import im.vector.ui.themes.ThemeUtils
+import kotlin.properties.Delegates
+
+
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+abstract class TroubleshootTest(val id: String, val title: String) {
+
+    enum class TestStatus {
+        NOT_STARTED,
+        RUNNING,
+        FAILED,
+        SUCCESS
+    }
+
+    var description: String? = null
+
+    var status: TestStatus by Delegates.observable(TestStatus.NOT_STARTED) { _, _, _ ->
+       statusListener?.invoke(this)
+    }
+
+    var statusListener : ((TroubleshootTest) -> Unit)? = null
+
+    abstract fun perform()
+
+    fun isFinihed() : Boolean {
+        return status == TroubleshootTest.TestStatus.FAILED || status == TroubleshootTest.TestStatus.SUCCESS
+    }
+
+    var quickFix: TroubleshootQuickFix? = null
+
+
+    abstract class TroubleshootQuickFix(val title: String) {
+        abstract fun doFix()
+    }
+}
+
+
+
+
+class NotificationTroubleshootRecyclerViewAdapter(tests: ArrayList<TroubleshootTest>) : RecyclerView.Adapter<NotificationTroubleshootRecyclerViewAdapter.ViewHolder>() {
```

Comment:
I prefer each class to be in its own file

Reply:
Done. Created a new package 'troubleshoot'